### ISSUE TITLE
Durable Functions called from Logic Apps sometimes stuck as pending.

### ISSUE THREAD

**Issue by kashimiz** [Azure/azure-functions-durable-extension#592]
Cross-posted from [StackOverflow](https://stackoverflow.com/questions/54362809/azure-logic-apps-take-hours-days-to-run-azure-functions):

> I have a set of logic apps that each call a set function apps which are run in parallel. 
Each logic app is triggered to start at a certain time during the night with all staggered an hour apart.
The Azure functions are written using the async pattern and call external APIs.
>
> **Problem** 
Sometimes the logic apps will run fine and complete their execution in a normal time period, and can do so for two or three days in a row. 
However sometimes they will take hours or days forcing me to cancel their run.
>
> ![Stalled Logic app executions](https://i.stack.imgur.com/e5WtI.png)
>
> Can any body shed any light on this might be happening?
>
>**Notes**
> 1. I'm using the latest nuget packages of the durable functions extension
> 2. When debugging the functions always complete in a timely fashion
> 3. I have noticed that the functions sometimes get stuck at pending.

**Comment by CundyTech** [comment]
Hi Katy, 

Thank you for picking this up. Please see below for information requested.

**Investigative information**

Durable Functions extension version: 
Function App version (1.0 or 2.0): **2.0**
Programming language used: **c#**

If deployed to Azure

Timeframe issue observed: **27/01/2019, 06:15 - 29/01/2019, 06:15 (I have then triggered from a logic app at 6:00am daily.**
Function App name: **AzureComputeAPI**
Function name(s): **VirtualMachinesListAsync, ResourceSkusListAsync, VirtualMachinesSizesListAsync and VirtualMachinesStatusListAsync**
Region: **North Europe**

Additional context

Development environment (ex. Visual Studio): **Visual Studio 2017**

**Comment by tohling** [comment]
@CundyTech, I checked our logs and there are seems to be no errors in your Function executions during the incident time frame. 

However, I do see a lot of warnings where many requests are being made to execute a list of Functions that are either disabled/missing inside this Function App between 01-27-2019 and 01-29-2019. All these requests failed because the Functions host process was unable to detect these Functions.

Are these requests expected? 

*ChargesOrchestrator (Request count: 19)
*MetricsOrchestrator (Request count: 197)
*llingOrchestrator (Request count: 23)
*etListOrchestrator (Request count: 232)
*GroupsOrchestrator (Request count: 118)
*csDefinitionsOrchestrator (Request count: 115)
*cesListOrchestrator (Request count: 141)
*ailsListOrchestrator (Request count: 197)
*etGetOrchestrator (Request count: 268)
*ceDefinitionsOrchestrator (Request count:107 )
*ageOrchestrator (Request count: 15)
*ationsOrchestrator (Request count: 17)

**Comment by cgillum** [comment]
FYI @glennamanns since this might be related to https://github.com/Azure/azure-functions-durable-extension/issues/577 (assuming these functions weren't legitimately disabled).

**Comment by CundyTech** [comment]
Hi @tohling, 

So these functions are my orchestrators, which I have not disabled and of course are needed to run my functions in an async way. However what is strange is they all have had there names left clipped. Did this happen as you copy and pasted or is perhaps why they can't be found?

For example

ationsOrchestrator
RIRecommendationsOrchestrator

**Comment by cgillum** [comment]
Hi @CundyTech. I looked into this further and it appears you have at least two function apps that are configured with the same storage account and task hub name:

* AzureConsumptionAPI
* AzureComputeAPI

This causes the two function apps to steal messages from each other, resulting in the "function not found" warnings mentioned previously.

Please see the Task Hubs documentation for more information: https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-task-hubs. Let us know if you have any further questions on this.

**Comment by CundyTech** [comment]
Thank you @cgillum. Just a side note. 

Wouldn't it be better for each new function it have it's own Task Hub by default as stated below. 

"Each function app has a separate task hub. If multiple function apps share a storage account, the storage account contains multiple task hubs. The following diagram illustrates one task hub per function app in shared and dedicated storage accounts."
[Task hubs in Durable Functions (Azure Functions)](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-task-hubs)

In my case the above statement was not true.

**Comment by cgillum** [comment]
@CundyTech yes, that would be much better. Unfortunately we didn't ship with this behavior, and making that change now would be breaking. We do have a bug tracking doing this as part of a breaking change release here: https://github.com/Azure/azure-functions-durable-extension/issues/383.

And you correctly point out that the documentation is misleading. I'll work on getting that updated. Thanks for pointing that out!
